async_test.go

```diff
@@ -136,3 +136,39 @@ func retrieveRows(rows *RowsExtended, ch chan string) {
 	ch <- s
 	close(ch)
 }
+
+func TestLongRunningAsyncQuery(t *testing.T) {
+	db := openDB(t)
+	defer db.Close()
+
+	ctx, _ := WithMultiStatement(context.Background(), 0)
+	query := "CALL SYSTEM$WAIT(50, 'SECONDS');use snowflake_sample_data"
+
+	rows, err := db.QueryContext(WithAsyncMode(ctx), query)
+	if err != nil {
+		t.Fatalf("failed to run a query. %v, err: %v", query, err)
+	}
+	defer rows.Close()
+	var v interface{}
+	i := 0
+	for {
```

Comment:
Can we write it as `for i := 0; rows.NextResultSet(); i++`, what do you think?

Reply:
This skips the first result set.

Reply:
Good catch! I fixed my code as well, if you find it more readable it would be great: https://github.com/snowflakedb/gosnowflake/pull/827/files#diff-9e7181917b4fbf7b4e6ca304941bafec645e1cd26eaef9a6c191f43c7166008aR70

Reply:
@sfc-gh-pfus sorry I missed this but I see your PR is already merged.